api/apps/api/test/integration/typeorm-export.repository.integration.e2e-spec.ts

```diff
@@ -0,0 +1,212 @@
+import { ClonePiece, ComponentId, ResourceKind } from '@marxan/cloning/domain';
+import { FixtureType } from '@marxan/utils/tests/fixture-type';
+import { Connection } from 'typeorm';
+import { v4 } from 'uuid';
+import { ResourceId } from '../../src/modules/clone/export';
+import { ExportEntity } from '../../src/modules/clone/export/adapters/entities/exports.api.entity';
+import { ComponentLocation } from '../../src/modules/clone/export/application/complete-piece.command';
+import { ExportRepository } from '../../src/modules/clone/export/application/export-repository.port';
+import { Export, ExportId } from '../../src/modules/clone/export/domain';
+import { bootstrapApplication } from '../utils/api-application';
+
+describe('Typeorm export repository', () => {
+  let fixtures: FixtureType<typeof getFixtures>;
+
+  beforeEach(async () => {
+    fixtures = await getFixtures();
+  }, 20000);
+
+  afterAll(async () => {
+    await fixtures.cleanup();
+  });
+
+  it('should expose methods for getting an export by id and storing exports', async () => {
+    await fixtures.GivenExportWasRequested();
+
+    const exportData = await fixtures.WhenReadingTheSavedExportFromRepository();
+
+    await fixtures.ThenExportDataShouldBeOk({
+      exportData,
+      componentsAreCompleted: false,
+    });
+  });
+
+  it('should update an export when a nested piece is completed', async () => {
+    await fixtures.GivenExportWasRequested();
+    await fixtures.WhenAComponentIsCompleted();
+    const exportData = await fixtures.WhenReadingTheSavedExportFromRepository();
+
+    await fixtures.ThenExportDataShouldBeOk({
+      exportData,
+      componentsAreCompleted: true,
+    });
+  });
+
+  it('should save entities working with transaction', async () => {
+    await fixtures.GivenExportWithMultipleComponentsWasRequested();
+    await fixtures.WhenAllComponentsAreCompletedConcurrentlyWithTransactions();
+
+    const exportData = await fixtures.WhenReadingTheSavedExportFromRepository();
+
+    fixtures.ThenAllExportComponentsShouldBeFinished({
+      exportData,
+    });
+  });
+});
+
+const getFixtures = async () => {
+  let exportId: ExportId;
+  let resourceId: ResourceId;
+  let componentId: ComponentId;
+  let componentLocationUri: string;
+  let componentLocationRelativePath: string;
+
+  const app = await bootstrapApplication();
+  const repo = app.get<ExportRepository>(ExportRepository);
+
+  return {
+    cleanup: async () => {
+      const connection = app.get<Connection>(Connection);
+      const exportRepo = connection.getRepository(ExportEntity);
+      const exportComponentRepo = connection.getRepository(
+        ExportComponentEntity,
+      );
+      const locationRepo = connection.getRepository(ComponentLocationEntity);
+
+      await locationRepo.delete({});
+      await exportComponentRepo.delete({});
+      await exportRepo.delete({});
```

Comment:
could be a good secret test of cascade delete (running only `await exportRepo.delete({});`)